src/renderer/viz/expressions/Image.js

```diff
@@ -80,43 +80,26 @@ export default class Image extends Base {
     }
 
     _preDraw (program, drawMetadata, gl) {
-        if (!this.init && this.canvas) {
-            this.init = true;
-            gl.activeTexture(gl.TEXTURE0 + drawMetadata.freeTexUnit);
+        if (!this.texture) {
             this.texture = gl.createTexture();
+        }
+
+        gl.activeTexture(gl.TEXTURE0 + drawMetadata.freeTexUnit);
+        gl.bindTexture(gl.TEXTURE_2D, this.texture);
+        gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, new Uint8Array(4));
+        gl.uniform1i(this._getBinding(program)._texLoc, drawMetadata.freeTexUnit);
+        drawMetadata.freeTexUnit++;
+
+        if (this._loaded) {
             gl.pixelStorei(gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL, false);
             gl.bindTexture(gl.TEXTURE_2D, this.texture);
-            gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, this.canvas);
-            gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR_MIPMAP_LINEAR);
-            gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.LINEAR);
+
+            this.image.width = this.image.width || 150;
```

Comment:
In the old `_getCanvasFromImage` a default 256 value is used for canvas size (and it was just using the original `img.width` and `img.height`). Does it make sense to use here that 256 / 2 = 128 size ?